### PR TITLE
DR-409 Aws gateway url added to integration tests

### PR DESCRIPTION
**Pull request by tamaragoldschmidt**
## Description
Updates to be able to construct the api-url based on the workspaced infrastructure and run the tests via a script
Rename the tests to be integration rather than e2e

## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [x] I have added tests to cover my changes
- [x] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: scripts/integration-tests.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+source ./scripts/functions/git-functions.sh
+
+# identify the workspace name and set TERRAFORM_WORKSPACE_NAME
+export_terraform_workspace_name
+
+# check export has been done
+EXPORTS_SET=0
+if [ -z "$TERRAFORM_WORKSPACE_NAME" ] ; then
+  echo Set TERRAFORM_WORKSPACE_NAME
```
Comment (timrickwood) [inline scripts/integration-tests.sh:17]: Debug code ?

File: scripts/integration-tests.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+if [ $EXPORTS_SET = 1 ] ; then
+  echo TERRAFORM_WORKSPACE_NAME exports not set
+  exit 1
+fi
+
+# install requirements
+echo "Installing requirements"
+pip install -r $APPLICATION_TEST_DIR/requirements.txt
+
+echo "Running tests for E2E api"
```
Comment (timrickwood) [inline scripts/integration-tests.sh:30]: Running  api integration tests - rather than E2E ?

File: tests/integration/steps/common_api_steps.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+from behave import given, then, step
+
+import requests
+from assertpy import assert_that
+
+
+@given("I send a request to the resource {resource_name}")
+def send_request(context, resource_name):
+    context.response = requests.get(context.URL + "/" + resource_name)
+    # assert_that((context.URL)+"/"+resource_name).is_equal_to("233")
```
Comment (timrickwood) [inline tests/integration/steps/common_api_steps.py:10]: Dead code

File: tests/integration/steps/simple_steps.py [hunk 4]
```diff
@@ -11,6 +11,7 @@ def impl_bk(context):
 
 @then("I receive a 200 status response")
 def get_bk(context):
+    # assert_that(context.response.status_code).is_equal_to(context.URL)
```
Comment (timrickwood) [inline tests/integration/steps/simple_steps.py:14]: dead code

File: scripts/integration-tests.sh (hunk 2 above)
Comment (tamaragoldschmidt) [reply scripts/integration-tests.sh:30]: updated

File: tests/integration/steps/common_api_steps.py (hunk 3 above)
Comment (tamaragoldschmidt) [reply tests/integration/steps/common_api_steps.py:10]: removed

File: tests/integration/steps/simple_steps.py (hunk 4 above)
Comment (tamaragoldschmidt) [reply tests/integration/steps/simple_steps.py:14]: removed

**Review by timrickwood — APPROVED** [review]

**Review by jope1 — APPROVED** [review]

**Pull request merged**